app/src/main/kotlin/batect/config/io/ConfigurationFile.kt

```diff
@@ -26,10 +27,21 @@ data class ConfigurationFile(
     val containers: Map<String, ContainerFromFile> = emptyMap()
 ) {
 
-    fun toConfiguration(pathResolver: PathResolver): Configuration = Configuration(
-        resolveProjectName(pathResolver),
-        TaskMap(tasks.map { (name, task) -> task.toTask(name) }),
-        ContainerMap(containers.map { (name, container) -> container.toContainer(name, pathResolver) }))
+    fun toConfiguration(pathResolver: PathResolver): Configuration {
+
+        return Configuration(
+            resolveProjectName(pathResolver),
+            TaskMap(tasks.map { (name, task) -> task.toTask(name) }),
+            ContainerMap(containers.map { (name, container) -> fromFileToContainerConfiguration(name, container, pathResolver) }))
+    }
+
+    private fun fromFileToContainerConfiguration(containerName: String, fileContainerConfig: ContainerFromFile, pathResolver: PathResolver): Container {
+        if (fileContainerConfig == null) {
```

Comment:
This is awful behaviour from Jackson... it really shouldn't be providing a null value here for something that can't be null. Not much we can do about that though, and it's already been reported at https://github.com/FasterXML/jackson-module-kotlin/issues/144. (This makes me wonder where else we need to cover this case...)

It also means the compiler gives a warning about this condition (https://travis-ci.org/charleskorn/batect/builds/398475902#L577), because it should theoretically never be true. Could you perhaps make `fileContainerConfig` nullable (ie. `ContainerFromFile?` rather than `ContainerFromFile`) so that the compiler warning goes away?